src/ast2ram/ClauseTranslator.h

```diff
@@ -32,17 +32,46 @@ namespace souffle::ast2ram {
 
 class TranslatorContext;
 
+/** Translation modes */
+enum TranslationMode {
+    DEFAULT,
+    // Subsumptive clause translation:
+    //
+    //   R(x0) <= R(x1) :- body.
+    //
+    // are translated to clauses of the format
+    //
+    //   R(x0) :- R(x0), R(x1), body.
+    //
+    // using different auxiliary tables (delta/new/reject/delete).
+    // There are different modes for translating a subsumptive clause
+    // (i.e. inside / outside of the fix-point).
+    //
+    SubsumeRNC,  // delete reject-R(x0) :- new-R(x0), R(x1), body. (inside fix-point)
+    SubsumeRNN,  // delete reject-R(x0) :- new-R(x0), R(x1), x0!=x1, body. (inside fix-point)
+    SubsumeDCD,  // delete delete-R(x0) :- R(x0), delta-R(x1), body. (inside fix-point)
```

Comment:
The three translation modes for the case 'inside fix-point' should be (in my understanding):
- SubsumeRNC,  // delete reject-R(x0) :- new-R(x0), R(x1), body. (inside fix-point)
- SubsumeRNN,  // delete reject-R(x0) :- new-R(x0), new-R(x1), x0!=x1, body. (inside fix-point)
- SubsumeDCN, // delete delete-R(x0) :- R(x0), new-R(x1), body. (inside fix-point)

I don't understand when SubsumeDCD would be useful.

Reply:
Thanks for the feedback. Let's merge the current version and do some subsequent PRs to fix subsumtive clauses and improve the code.